### PR TITLE
[12_3_X] Added the switch for the inclusion of PS energy

### PR DESCRIPTION
**Pull request by ram1123**
#### PR description:

EGamma regression can be trained with and without PS energy taking into consideration. So, we added a switch for this while performing the evaluation in the CMSSW.

***This PR is relevant ONLY for HLT supercluster regression, and that it does not effect the offline egamma regression.***

#### PR validation:

We run the code with the current version of update and it worked as expected.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

https://github.com/cms-sw/cmssw/pull/37491

We need this update from 12_3_X onwards.

### CODE REVIEW

**Comment by perrotta** [comment]
backport of https://github.com/cms-sw/cmssw/pull/37491

**Comment by francescobrivio** [comment]
@ram1123 please include "[12_3_X]" in the title of the PR

**Comment by francescobrivio** [comment]
urgent

 - This is potentially needed for 12_3_0, but it also requires an update to the HLT GT which has not been officially requested yet (see https://github.com/cms-sw/cmssw/pull/37491#issuecomment-1091946705). I'll bring this up at the ORP tomorrow.

File: RecoEcal/EgammaClusterAlgos/src/SCEnergyCorrectorSemiParm.cc [hunk 1]
```diff
@@ -51,6 +51,7 @@ SCEnergyCorrectorSemiParm::SCEnergyCorrectorSemiParm()
       caloGeom_(nullptr),
       isHLT_(false),
       isPhaseII_(false),
+      regTrainedWithPS_(true),
```
Comment (francescobrivio) [inline RecoEcal/EgammaClusterAlgos/src/SCEnergyCorrectorSemiParm.cc:54]: @ram1123 please change this to `false`, see discussion at https://github.com/cms-sw/cmssw/pull/37491#pullrequestreview-935418113

**Review by francescobrivio — CHANGES_REQUESTED** [review]

**Comment by qliphy** [comment]
please test

**Comment by clacaputo** [comment]
+reconstruction

- backport https://github.com/cms-sw/cmssw/pull/37491
- PR relevant only for HLT supercluster regression
- boolean set to false and doesn't change the previous behaviour see https://github.com/cms-sw/cmssw/pull/37491#pullrequestreview-935418113

**Comment by perrotta** [comment]
+1
- The fix is simple, and it is an exact backport of #37491: let have it merged even if the IB tests of the master version haven't been concluded yet

**Pull request merged**